pkg/profiler/cpu/metrics.go

```diff
@@ -60,5 +61,13 @@ func newMetrics(reg prometheus.Registerer) *metrics {
 				NativeHistogramBucketFactor: 1.1,
 			},
 		),
+		obtainMapMetrics: promauto.With(reg).NewGaugeVec(
```

Comment:
Is it possible that we add/remove new maps at runtime? If so then we should rather implement a Prometheus collector, rather than setting the metrics. A collector instead collects the metrics at scrape time, rather than setting them synchronously like here. If maps might come and go over time we should implement a collector (the implemented functionality would still be super close to what was implemented in this PR, just thinking about future-proofing this).